src/components/swap/SwapModalFooter.tsx

```diff
@@ -1,30 +1,44 @@
-import { Trans } from '@lingui/macro'
-import { TraceEvent } from '@uniswap/analytics'
-import { BrowserEvent, InterfaceElementName, SwapEventName } from '@uniswap/analytics-events'
+import { t, Trans } from '@lingui/macro'
+import { sendAnalyticsEvent, TraceEvent } from '@uniswap/analytics'
+import {
+  BrowserEvent,
+  InterfaceElementName,
+  SwapEventName,
+  SwapPriceUpdateUserResponse,
+} from '@uniswap/analytics-events'
+import { formatCurrencyAmount, formatPriceImpact, NumberType } from '@uniswap/conedison/format'
 import { Currency, Percent, TradeType } from '@uniswap/sdk-core'
+import { useWeb3React } from '@web3-react/core'
+import Column from 'components/Column'
 import useTransactionDeadline from 'hooks/useTransactionDeadline'
+import useNativeCurrency from 'lib/hooks/useNativeCurrency'
 import {
   formatPercentInBasisPointsNumber,
   formatPercentNumber,
   formatToDecimal,
   getDurationFromDateMilliseconds,
   getDurationUntilTimestampSeconds,
+  getPriceUpdateBasisPoints,
   getTokenAddress,
 } from 'lib/utils/analytics'
-import { ReactNode } from 'react'
-import { Text } from 'rebass'
+import { ReactNode, useEffect, useMemo, useState } from 'react'
+import { AlertTriangle } from 'react-feather'
 import { RouterPreference } from 'state/routing/slice'
 import { InterfaceTrade } from 'state/routing/types'
 import { useRouterPreference, useUserSlippageTolerance } from 'state/user/hooks'
+import styled, { useTheme } from 'styled-components/macro'
+import { ThemedText } from 'theme'
+import { formatTransactionAmount, priceToPreciseFloat } from 'utils/formatNumbers'
 import getRoutingDiagramEntries, { RoutingDiagramEntry } from 'utils/getRoutingDiagramEntries'
-import { computeRealizedPriceImpact } from 'utils/prices'
+import { computeRealizedPriceImpact, getPriceImpactWarning } from 'utils/prices'
 
-import { ButtonError } from '../Button'
-import { AutoRow } from '../Row'
-import { SwapCallbackError } from './styleds'
+import { ButtonError, SmallButtonPrimary } from '../Button'
+import { AutoRow, RowBetween, RowFixed } from '../Row'
+import { SwapCallbackError, SwapShowAcceptChanges } from './styleds'
+import { SwapModalDetailRow } from './SwapModalDetailRow'
 
 interface AnalyticsEventProps {
-  trade: InterfaceTrade<Currency, Currency, TradeType>
+  trade: InterfaceTrade<Currency, Currency, TradeType> | undefined
```

Comment:
why are u allowing `undefined` trades here now

Reply:
the UI was flickering because trade was briefly undefined, i think because the quote was reloading. this only happened when you waited long enough or switched focus back to the window.

Reply:
ah yes that is a bug. can you rebase this after i land https://github.com/Uniswap/interface/pull/6499 ? trade should not be undefined once this fix goes on.

Reply:
If it wasn't undefined previously, it means that the UI was flickering already before you started working on the PR? In that case, I would consider that as non-issue and just move on without `undefined`. If after working here, there's a bug in the code to be fixed instead.